src/EditorFeatures/Test/Workspaces/TestWorkspaceFactory_XmlConsumption.cs

```diff
@@ -139,7 +139,14 @@ public static TestWorkspace CreateWorkspace(string xmlDefinition, bool completed
 
             for (int i = 1; i < submissions.Count; i++)
             {
-                workspace.OnProjectReferenceAdded(submissions[i].Id, new ProjectReference(submissions[i - 1].Id));
+                for (int j = i - 1; j >= 0; j--)
```

Comment:
You now have to add all references versus just the immediate chain now? Doesn't this invalidate the existence of the algorithm you're fixing up in the first place?